### PR TITLE
UCS/ARCH: Add Zhaoxin cpu detection

### PR DESCRIPTION
**Pull request by zzjianhui**
Add support for Zhaoxin processors. 

Including three micro-architectures: Zhangjiang, Wudaokou, Lujiazui.

Added support for three processors: ZX-C, KX5000, KX6000.

### CODE REVIEW

**Comment by zzjianhui** [comment]
> 
> 
> Can one of the admins verify this patch?

Yes

**Comment by brminich** [comment]
ok to test

**Comment by shamisp** [comment]
@zzjianhui - thanks for the patch ! What is your affiliation ? I just want to make sure that your organization have the CLA signed. Also the copyright header has to be updated. thanks !

**Review by shamisp — CHANGES_REQUESTED** [review]
Waiting for CLA confirmation

**Comment by zzjianhui** [comment]
@shamisp How to get CLA signed, can you tell me the process, thanks

**Comment by shamisp** [comment]
@zzjianhui Print coproporate or individual CLA from here https://www.openucx.org/license/.
Sign and email the copy to <admin at ucfconsortium.org>
We also ask mail the hard copy but we can merge as soon as we have digital copy. Once you email it, please let me know on this thread. Thanks !

**Comment by zzjianhui** [comment]
@shamisp Mailing List cannot be opened. Can you give me email address?

**Comment by zzjianhui** [comment]
@shamisp I send CLA to your email address

**Comment by zzjianhui** [comment]
@shamisp I re-sent an email to you and added the CLA attachment. Thanks

**Comment by shamisp** [comment]
@zzjianhui we have the received the form. I'm waiting for our admin to review it and countersign. thanks !

**Comment by shamisp** [comment]
Thanks for signing the CLA.

File: src/ucs/arch/x86_64/cpu.c [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 #define X86_CPUID_GENUINEINTEL    "GenuntelineI" /* GenuineIntel in magic notation */
 #define X86_CPUID_AUTHENTICAMD    "AuthcAMDenti" /* AuthenticAMD in magic notation */
+#define X86_CPUID_CENTAURHAULS    "CentaulsaurH" /* CentaurHauls in magic notation */
```
Comment (shamisp) [inline src/ucs/arch/x86_64/cpu.c:22]: Please add your organization copyright to the top of the file. The same relevant for the rest of updated files.

**Review by shamisp — CHANGES_REQUESTED** [review]

File: src/tools/info/sys_info.c [hunk 2]
```diff
@@ -29,7 +29,10 @@ static const char* cpu_model_names[] = {
     [UCS_CPU_MODEL_INTEL_SKYLAKE]     = "Skylake",
     [UCS_CPU_MODEL_ARM_AARCH64]       = "ARM 64-bit",
     [UCS_CPU_MODEL_AMD_NAPLES]        = "Naples",
-    [UCS_CPU_MODEL_AMD_ROME]          = "Rome"
+    [UCS_CPU_MODEL_AMD_ROME]          = "Rome",
+    [UCS_CPU_MODEL_ZHAOXIN_Zhangjiang]= "Zhangjiang",
+    [UCS_CPU_MODEL_ZHAOXIN_Wudaokou]  = "Wudaokou",
+    [UCS_CPU_MODEL_ZHAOXIN_Lujiazui]  = "Lujiazui"
```
Comment (yosefe) [inline src/tools/info/sys_info.c:35]: 1. use uppercase: UCS_CPU_MODEL_ZHAOXIN_ZHANGJIANG
2. space before = on line 33

File: src/ucs/arch/x86_64/cpu.c [hunk 3]
```diff
@@ -334,53 +336,72 @@ ucs_cpu_model_t ucs_arch_get_cpu_model()
...
-       case 0x56:
-           return UCS_CPU_MODEL_INTEL_BROADWELL;
-       case 0x5e:
-       case 0x4e:
-       case 0x55:
-           return UCS_CPU_MODEL_INTEL_SKYLAKE;
-       }
-    }
+    if (ucs_arch_get_cpu_vendor() == UCS_CPU_VENDOR_ZHAOXIN){
+        if (family == 0x06){
```
Comment (yosefe) [inline src/ucs/arch/x86_64/cpu.c:344]: space before {

File: src/ucs/arch/x86_64/cpu.c [hunk 4]
```diff
@@ -334,53 +336,72 @@ ucs_cpu_model_t ucs_arch_get_cpu_model()
...
-       case 0x4f:
-       case 0x56:
-           return UCS_CPU_MODEL_INTEL_BROADWELL;
-       case 0x5e:
-       case 0x4e:
-       case 0x55:
-           return UCS_CPU_MODEL_INTEL_SKYLAKE;
-       }
-    }
+    if (ucs_arch_get_cpu_vendor() == UCS_CPU_VENDOR_ZHAOXIN){
```
Comment (yosefe) [inline src/ucs/arch/x86_64/cpu.c:343]: space before {

File: src/ucs/arch/x86_64/cpu.c (hunk 1 above)
Comment (zzjianhui) [reply src/ucs/arch/x86_64/cpu.c:23]: fixed

File: src/tools/info/sys_info.c (hunk 2 above)
Comment (zzjianhui) [reply src/tools/info/sys_info.c]: fixed

File: src/ucs/arch/x86_64/cpu.c (hunk 3 above)
Comment (zzjianhui) [reply src/ucs/arch/x86_64/cpu.c]: fixed

File: src/ucs/arch/x86_64/cpu.c (hunk 4 above)
Comment (zzjianhui) [reply src/ucs/arch/x86_64/cpu.c]: fixed

File: src/tools/info/sys_info.c [hunk 5]
```diff
@@ -30,9 +30,9 @@ static const char* cpu_model_names[] = {
     [UCS_CPU_MODEL_ARM_AARCH64]       = "ARM 64-bit",
     [UCS_CPU_MODEL_AMD_NAPLES]        = "Naples",
     [UCS_CPU_MODEL_AMD_ROME]          = "Rome",
-    [UCS_CPU_MODEL_ZHAOXIN_Zhangjiang]= "Zhangjiang",
-    [UCS_CPU_MODEL_ZHAOXIN_Wudaokou]  = "Wudaokou",
-    [UCS_CPU_MODEL_ZHAOXIN_Lujiazui]  = "Lujiazui"
+    [UCS_CPU_MODEL_ZHAOXIN_ZHANGJIANG] = "Zhangjiang",
+    [UCS_CPU_MODEL_ZHAOXIN_WUDAOKOU]  = "Wudaokou",
+    [UCS_CPU_MODEL_ZHAOXIN_LUJIAZUI]  = "Lujiazui"
```
Comment (yosefe) [inline src/tools/info/sys_info.c:35]: pls indent all other entries as well, so the '=' will be con same column

Comment (zzjianhui) [reply src/tools/info/sys_info.c]: fixed

Comment (zzjianhui) [reply src/tools/info/sys_info.c]: @yosefe i only add space, why did it report an error?

Comment (brminich) [reply src/tools/info/sys_info.c]: error is unrelated, I just restarted failed jobs

Comment (zzjianhui) [reply src/tools/info/sys_info.c]: @brminich Still reporting an error

Comment (zzjianhui) [reply src/tools/info/sys_info.c]: @brminich Still reporting an error, Is it a problem with the test script?

Comment (brminich) [reply src/tools/info/sys_info.c]: unrelated issue, which should be fixed by #5949, let's restart testing when it is merged

File: src/ucs/arch/cpu.h [hunk 6]
```diff
@@ -30,6 +30,9 @@ typedef enum ucs_cpu_model {
     UCS_CPU_MODEL_ARM_AARCH64,
     UCS_CPU_MODEL_AMD_NAPLES,
     UCS_CPU_MODEL_AMD_ROME,
+    UCS_CPU_MODEL_ZHAOXIN_ZHANGJIANG,
```
Comment (shamisp) [inline src/ucs/arch/cpu.h:33]: Please add your copyright to the file

File: src/ucs/arch/cpu.c [hunk 7]
```diff
@@ -63,6 +63,10 @@ const ucs_cpu_builtin_memcpy_t ucs_cpu_builtin_memcpy[UCS_CPU_VENDOR_LAST] = {
     [UCS_CPU_VENDOR_FUJITSU_ARM] = {
         .min = UCS_MEMUNITS_INF,
         .max = UCS_MEMUNITS_INF
+    },
```
Comment (shamisp) [inline src/ucs/arch/cpu.c:66]: Please add your copyright to the file.

File: src/tools/info/sys_info.c [hunk 8]
```diff
@@ -19,17 +19,20 @@
...
-    [UCS_CPU_MODEL_INTEL_SANDYBRIDGE] = "SandyBridge",
-    [UCS_CPU_MODEL_INTEL_NEHALEM]     = "Nehalem",
-    [UCS_CPU_MODEL_INTEL_WESTMERE]    = "Westmere",
-    [UCS_CPU_MODEL_INTEL_HASWELL]     = "Haswell",
-    [UCS_CPU_MODEL_INTEL_BROADWELL]   = "Broadwell",
-    [UCS_CPU_MODEL_INTEL_SKYLAKE]     = "Skylake",
-    [UCS_CPU_MODEL_ARM_AARCH64]       = "ARM 64-bit",
-    [UCS_CPU_MODEL_AMD_NAPLES]        = "Naples",
-    [UCS_CPU_MODEL_AMD_ROME]          = "Rome"
+    [UCS_CPU_MODEL_UNKNOWN]            = "unknown",
```
Comment (shamisp) [inline src/tools/info/sys_info.c:22]: Please add your copyright to the file.

**Review by shamisp — CHANGES_REQUESTED** [review]

File: src/ucs/arch/cpu.h (hunk 6 above)
Comment (zzjianhui) [reply src/ucs/arch/cpu.h:34]: fixed

File: src/ucs/arch/cpu.c (hunk 7 above)
Comment (zzjianhui) [reply src/ucs/arch/cpu.c:67]: fixed

File: src/tools/info/sys_info.c (hunk 8 above)
Comment (zzjianhui) [reply src/tools/info/sys_info.c:23]: fixed

**Review by shamisp — APPROVED** [review]

**Review by yosefe — APPROVED** [review]
@shamisp are we good with CLA?

**Comment by shamisp** [comment]
> @shamisp are we good with CLA?

yes

**Pull request merged**